packages/libsql/src/lib/Category.ts

```diff
@@ -0,0 +1,417 @@
+import type {
+  IEventData,
+  StreamToken,
+  SyncResult,
+  ITimelineEvent,
+  TokenAndState,
+  ICodec,
+} from "@equinox-js/core"
+import * as Equinox from "@equinox-js/core"
+import * as Token from "./Token.js"
+import * as Read from "./Read.js"
+import { trace } from "@opentelemetry/api"
+import { Format, LibSqlReader, LibSqlWriter } from "./LibSqlClient.js"
+import { CachingCategory, ICachingStrategy, IReloadableCategory, Tags } from "@equinox-js/core"
+import { Client, Transaction } from "@libsql/client"
+import { randomUUID } from "crypto"
+
+const keepMap = Equinox.Internal.keepMap
+
+type GatewaySyncResult = { type: "Written"; token: StreamToken } | { type: "ConflictUnknown" }
```

Comment:
gateway is very old naming re esdb and trying to shoehorn in POESA naming
there is client, internalcategory and caterogy in play, maybe than naming can realign with that?
(PR accepted for doign that in upstream too)